autorest/azure/async.go

```diff
@@ -38,6 +39,8 @@ const (
 	operationSucceeded  string = "Succeeded"
 )
 
+var pollingCodes = [...]int{http.StatusAccepted, http.StatusCreated, http.StatusOK}
```

Comment:
🤔 is it also worth retrying/polling for `500 InternalServerError` and `429 Too Many Requests` - or is that handled elsewhere?

Reply:
We will still retry for those error codes (i.e. Done() will return false).  In this context these codes are used to determine if we should update the internal polling status or not.